models/xlat_test.xlat

```diff
@@ -0,0 +1,58 @@
+xlat_data = {}
+
+xlat_data.translate_type = function(field, value, dir)
+    local value1 = value
+    if field == 'type' then
+        value1 = tonumber (value)
+        print("field " .. field)
+        print("value1 " .. value1)
+        print("dir " .. dir)
+        if dir == 'out' then
+            if value == '1' then
+                value1 = '3'
+            else
+                value1 = '4'
+            end
+            print("value1 " .. value1)
+        else
+            if value == '3' then
+                value1 = '1'
+            else
+                value1 = '2'
+            end
+        end
+    end
+    return value1
+end
+
+return {
+    {
+        external = '/xlat-test/xlat-animals',
+        internal = '/test/animals/animal',
+    },
+    {
+        external = '/xlat-test/xlat-animals/xlat-animal',
+        internal = '/test/animals/animal',
+    },
+    {
+        external = '/xlat-test/xlat-animals/xlat-animal/*/name',
+        internal = '/test/animals/animal/*/name',
+    },
+    {
+        external = '/xlat-test/xlat-animals/xlat-animal/*/type',
+        internal = '/test/animals/animal/*/type',
+        data_function = 'xlat_data.translate_type',
```

Comment:
why is this a string that names the function?

It is possible to reference a function directly in the C code, so it should be possible to pass the actual function here

Reply:
There is no great reason to make it a function. It would complicate the parsing routine in schema.c as at the moment the return is composed of key/value pairs, all of which are strings.

Reply:
Capturing the string leaves the user with needing to make functions (or their containing structs) global and not-clashing, which seems problematic to manage when this is scaled to many models and/or we have extensions etc

Reply:
You should be able to use these combined to capture the function reference and access it later:
http://www.lua.org/manual/5.1/manual.html#luaL_ref
http://www.lua.org/manual/5.1/manual.html#3.5

Reply:
Given how tricky LUA is this will take 1 or 2 days to implement. Your references are not very specific. Have you got a real coding example.

Reply:
Implemented in the next version of the patch